toolchain/check/lexical_lookup.h

```diff
@@ -28,12 +28,20 @@ class LexicalLookup {
     ScopeIndex scope_index;
   };
 
+  // A lookup result that has been temporarily removed from scope.
+  struct SuspendedResult {
+    // The lookup index.
+    size_t index;
```

Comment:
```suggestion
    int32_t index;
```

Per below common on GetLookupIndex

Reply:
I'm inclined to go with `uint32_t` here, and an assert in `Suspend` that it fits (which should never fail because a converted `NameId` can only barely exceed the range of `int32_t`).

Reply:
Changed to `uint32_t`.

Reply:
Per above, I'd suggest size_t or int32_t. My leaning would be away from uint32_t per [style](https://google.github.io/styleguide/cppguide.html#Integer_Types): "You should not use the unsigned integer types such as uint32_t, unless there is a valid reason such as representing a bit pattern rather than a number, or you need defined overflow modulo 2^N. In particular, do not use unsigned types to say a number will never be negative."

But maybe there's a specific intent here that I've missed.

Reply:
I've added a comment to make sure the intent is documented.